docs/ci-cd-environment/machine-types.md

```diff
@@ -69,21 +73,170 @@ Linux machine types can be paired with [Ubuntu 18.04 image][ubuntu1804], [Ubuntu
      8
   </td>
   <td>
-     16
+     16 GB
   </td>
   <td>
-     45
+     45 GB
   </td>
 </tr>
 </tbody>
 </table>
 
-Implementation of `e1` series of machine types:
+Implementation details:
+
+1. `Virtual CPU` hyperthreaded on a 3.4GHz Max Turbo 4.0GHz Intel® Core™ i7.
+2. `Memory` is implemented as DDR4 RAM.
+3. `Disk` is implemented as RAM drive backed by DDR4 RAM.
+
+### E2 Generation
+
+The `e2` series machines offer a balance of compute, memory, and cost. It is a good choice for most applications.
+
+<table style="background-color: rgb(255, 255, 255);">
+<thead>
+<tr>
+  <td>
+    Machine name
+  </td>
+  <td>
+    Virtual CPUs <sup>1</sup>
+  </td>
+  <td>
+    Memory <sup>2</sup>
+  </td>
+  <td>
+    Disk <sup>3</sup>
+  </td>
+</tr>
+</thead>
+<tbody>
+<tr>
+  <td>
+    e2-standard-2
+  </td>
+  <td>
+     2
+  </td>
+  <td>
+     8 GB
+  </td>
+  <td>
+     55 GB
+  </td>
+</tr>
+<tr>
+  <td>
+    e2-standard-4
+  </td>
+  <td>
+     4
+  </td>
+  <td>
+     16 GB
+  </td>
+  <td>
+     75 GB
+  </td>
+</tr>
+<tr>
+  <td>
+     e2-standard-8
+  </td>
+  <td>
+     8
+  </td>
+  <td>
+     32 GB
+  </td>
+  <td>
+     100 GB
+  </td>
+</tr>
+</tbody>
+</table>
+
+Implementation details:
+
+1. `Virtual CPU` hyperthreaded on a 3.6GHZ [AMD Ryzen 5 3600](https://www.amd.com/en/product/8456).
+2. `Memory` is implemented as DDR4 RAM.
+3. `Disk` is implemented as RAM drive backed by DDR4 RAM.
+
+`e2` machines are currently only available on our [enterprise plan](https://semaphoreci.com/pricing).
+
+### F1 Generation
+
+The `f1` series machines are built for compute-intensive jobs, these machines deliver highest performance.. 
```

Comment:
deliver the highest performance.

there are two dots at the end of `performance..`